tests/handlers/aws/test_logstash_output.py

```diff
@@ -118,13 +123,77 @@ def tearDown(self) -> None:
             self.logs_client, group_name=self.group_name, stream_name=self.stream_name
         )
 
         ctx = ContextMock(TIMEOUT_15m)
         handler(event_cloudwatch_logs, ctx)  # type: ignore
-        # test new input => output to stdout
 
         msgs = self.logstash.get_messages()
         assert len(msgs) == 2
+
+        assert msgs[0]["fields"]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
+        assert msgs[1]["fields"]["message"] == self.fixtures["cw_log_2"].rstrip("\n")
+
+    def test_failure_sending_messages(self) -> None:
+        os.environ["S3_CONFIG_FILE"] = _prepare_config_file(
+            self,
+            "config.yaml",
+            # NOTE: using a bogus URL to mimic sending failure
+            dict(CloudwatchLogStreamARN=self.cloudwatch_group_arn, LogstashURL="http://fake.url"),
+            "folder/config2.yaml",
+        )
+
+        event_cloudwatch_logs, event_ids_cloudwatch_logs = _logs_retrieve_event_from_cloudwatch_logs(
+            self.logs_client, group_name=self.group_name, stream_name=self.stream_name
+        )
+
+        ctx = ContextMock(TIMEOUT_15m)
+        handler(event_cloudwatch_logs, ctx)  # type: ignore
+
+        msgs = self.logstash.get_messages()
+        assert len(msgs) == 0
+
+        messages = _sqs_get_messages(self.sqs_client, os.environ["SQS_REPLAY_URL"])
+        assert len(messages) == 2
+
+        event1 = json_parser(messages[0]["Body"])
+        assert event1["event_payload"]["fields"]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
+
+        event2 = json_parser(messages[1]["Body"])
+        assert event2["event_payload"]["fields"]["message"] == self.fixtures["cw_log_2"].rstrip("\n")
+
+    def test_send_timeout(self) -> None:
+        """
+        This test verify that if a timeout happens in progress events are correctly sent to the continuing queue.
+        """
+        os.environ["S3_CONFIG_FILE"] = _prepare_config_file(
+            self,
+            "config.yaml",
+            dict(CloudwatchLogStreamARN=self.cloudwatch_group_arn, LogstashURL=self.logstash.get_url()),
+            "folder/config2.yaml",
+        )
+
+        event_cloudwatch_logs, event_ids_cloudwatch_logs = _logs_retrieve_event_from_cloudwatch_logs(
+            self.logs_client, group_name=self.group_name, stream_name=self.stream_name
+        )
+
+        # NOTE: 0 triggers sending to the continuing queue as it mimics imminent timeout
+        ctx = ContextMock(0)
+        result = handler(event_cloudwatch_logs, ctx)  # type: ignore
+        assert result == "continuing"
+
+        msgs = self.logstash.get_messages()
+        assert len(msgs) == 1
+        assert msgs[0]["fields"]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
+
+        messages = _sqs_get_messages(self.sqs_client, os.environ["SQS_CONTINUE_URL"])
+        assert messages[0]["Body"] == self.fixtures["cw_log_1"] + self.fixtures["cw_log_2"]
```

Comment:
I would trigger the lambda again and assure now only `self.fixtures["cw_log_2"]` will be sent to logstash, meaning the continuation from the last processed offset is working properly